apps/web/pages/_error.tsx

```diff
@@ -70,3 +115,43 @@ class ErrorPage extends React.Component<ErrorPageProps> {
 }
 
 export default withApollo(ErrorPage)
+
+export type RedirectPropsContext<Context> = Context & {
+  apolloClient: ApolloClient<NormalizedCacheObject>
+  locale: string
+}
+
+export interface RedirectProps {
+  pageType: string
+  page: Article | ArticleCategory
+}
+
+const getRedirectProps = async (
+  slug: string,
+  { apolloClient, locale }: RedirectPropsContext<NextPageContext>,
```

Comment:
Do we need to use NextPageContext here, can't we just use an standard interface and pass the required arguments from ErrorPage.getInitialProps